vim-patch:8.2.{265,2761,2764}

janlazo opened the pull request.

janlazo committed:
vim-patch:8.2.2761: using "syn include" does not work properly

Problem:    Using "syn include" does not work properly.
Solution:   Don't add current_syn_inc_tag to topgrp. (Jaehwang Jerry Jung,
            closes vim/vim#8104)
https://github.com/vim/vim/commit/2e240bd428c0033d16f201d7f837636412358199

Cherry-pick AssertHighlightGroups from patch v8.2.2068.

janlazo committed:
ex_cmds: port :eval

Cherry-picked from patch v8.1.1807.
Required for patch v8.2.2761.

janlazo committed:
vim-patch:8.2.0265: "eval" after "if 0" doesn't check for following command

Problem:    "eval" after "if 0" doesn't check for following command.
Solution:   Add "eval" to list of commands that check for a following command.
            (closes vim/vim#5640)
https://github.com/vim/vim/commit/a76b31542e1d83b7d2fe7378439912e800a4d0f6

tomtomjhj commented:
[8.2.2764](https://github.com/vim/vim/commit/b47bed2f7ada4dfae78f76f27473b83507e40315) would fix the [1 byte memory leak](https://github.com/neovim/neovim/pull/14367/checks?check_run_id=2348458620#step:11:7719).

janlazo committed:
vim-patch:8.2.2764: memory leak when default function argument is allocated

Problem:    Memory leak when default function argument is allocated.
Solution:   Free the expression result.
https://github.com/vim/vim/commit/b47bed2f7ada4dfae78f76f27473b83507e40315
